pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1952,6 +1952,7 @@ which makes the code also more readable.
                 <value>
                     <![CDATA[
 //VariableInitializer
+/../..[(Type)]
 //PrimaryExpression[not(PrimarySuffix)]
 [not(ancestor::ArgumentList)]
 /PrimaryPrefix/AllocationExpression[ClassOrInterfaceType[@AnonymousClass='false']/TypeArguments//ReferenceType[not(.//TypeArguments)]]
```

Comment:
Could `TypeArguments//ReferenceType[not(.//TypeArguments)]]` be replaced by `TypeArguments[@Diamond="false"]`? That would avoid the `//` check as well. Again, I've not tested

That could be applied to the second branch of the expression, below

Reply:
```
private WeakReference<Class<?>> typeReference;
typeReference = new WeakReference<Class<?>>(type); //false positive
```